atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/AtlasDbTestCase.java

```diff
@@ -92,7 +89,24 @@ public void setUp() throws Exception {
     }
 
     private void setUpTransactionManagers() {
-        serializableTxManager = wrapTestTransactionManager(new TestTransactionManagerImpl(
+        serializableTxManager = constructTestTransactionManager();
+
+        txManager = new CachingTestTransactionManager(serializableTxManager);
+    }
+
+    protected TrackingKeyValueService setUpKeyValueService() {
+        KeyValueService kvs = wrapKeyValueService(getBaseKeyValueService());
+        keyValueServiceWithStats = new StatsTrackingKeyValueService(kvs);
+        keyValueService = spy(new TrackingKeyValueService(keyValueServiceWithStats));
```

Comment:
this is a bad refactoring/at the very least it's confusing.

Have this method take in the `baseKeyValueService`, and then *don't* set the field in this method, and return the wrapped kvs

Reply:
Wrapping is now also obsolete since it is done differently. Removed `keyValueServiceWithStats`.